README.md

```diff
@@ -1,57 +1,97 @@
 # Website Digital Product School
 
-# Prerequesites
+# Frontend
 
-## Node and npm
+## Prerequesites
 
-` Install node and npm `
+### Node and npm
 
-## Gatsby
+`Install node and npm`
+
+### Gatsby
 
 `npm install -g gatsby-cli`
 
+## Context
+
+The gatsby frontend resides currently in the root directory. The intention is to move it to e `frontend/` folder.
+
+## Setting Up
+
+1. Run `npm install` in the root dir
+
+## Development
+
+1. Run `npm run start`
+2. Any environemnt variables go into `.env.development` for local development and into `cloudbuild-prod|staging.yaml` for prod / staging. This is a bit inconsistent, it might be best to move those from `cloudbuild-prod|staging.yaml` in `.env.staging` and `.env.production`, but yolo unless you wanna fix it.
+
+## Testing
+
+???
+
+## Deployment
+
+Any push to staging branch is being deployed via `cloudbuild-staging.yaml` and any push to master branch is being deplopyed via `cloudbuild-prod.yaml`.
+
+# Services / Cloud Functions
+
+## Context
+
+This is work in progress and we're still figuring out what's the best way to develop cloud functions. You can find them under `funtions/`.
+
+## Setting up
 
-# Setting Up
-A step by step guide that tell you how to get the development and production environment running
+Navigate to `functions/` and run `npm run install`.
 
-After cloning the repository install the package dependencies using:
+## Development
 
-```
-npm install
-```
+1. Running `npm run serve` will start the functions locally. However, those functions connect to `firestore` on the cloud. The other option would be to start `firestore` emulator as well, but for now we like the cloud one, because it provides nice UI and is less hassle to setup. Those functions would connect against our staging `cloud storage` as well.
 
-# Development
+2. In order to authenticate against `firestore` and `cloud storage`, you need to go to the dps-website-staging project and download the key for `dps-website-staging-0@appspot.gserviceaccount.com` and put it under `functions/` with name `dps-website-staging.firebase.key.json`.
 
-Make sure you are in the root directory
-Start the development server using
-```
-gatsby develop
-```
-and visit
-```
-http://localhost:8000/
-```
+3. Now you can experiment against the functions that run locally and are accessible via localhost, but they access our staging database & storage.
```

Comment:
What happens with Functions with a trigger like a new entry in the database? 
Do they run locally and in staging then? Or is only one triggered? 🤔

Reply:
I think if we want local triggers we have to use emulators. I guess for now staging only is the way to test or write tests